tex/gregoriotex.lua

```diff
@@ -1237,101 +1174,80 @@ local function locate_file(filename)
+                     string.find(os.getenv("PATH"),";",1,true))
+  if onwindows then
+    sep = "\\"
+  else
+    sep = "/"
+  end
+    
+  if string.match(gabc_file, "[#%%]") then
     err("GABC filename contains invalid character(s): # %%\n"
-        .."Rename the file and retry: %s", input_file)
+        .."Rename the file and retry: %s", gabc_file)
   end
-  local has_extention = false
-  local file_dir,input_name
+  local gabc_dir, base
   local extensions = {['gabc']=true, ['gtex']=true, ['tex']=true}
-  if extensions[string.match(input_file, "([^%.\\/]*)$")] then
-    has_extention = true
-  end
-  if has_extention then
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
+  if extensions[string.match(gabc_file, "([^%.\\/]*)$")] then
+    gabc_dir, base = string.match(gabc_file, "(.-)([^\\/]-)%.?[^%.\\/]*$")
   else
-    file_dir,input_name = string.match(input_file, "(.-)([^\\/]*)$")
-  end
-
-  local cleaned_filename = input_name:gsub("[%s%+%&%*%?$@:;!\"\'`]", "-")
-  local gabc_filename = string.format("%s%s.gabc", file_dir, input_name)
-  local gabc_file = locate_file(gabc_filename)
-  local gtex_filename = string.format("%s%s-%s.gtex", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  local gtex_file = locate_file(gtex_filename)
-  local glog_file = string.format("%s%s-%s.glog", file_dir, cleaned_filename,
-      internalversion:gsub("%.", "_"))
-  if not gtex_file then
-    clean_old_gtex_files(file_dir..cleaned_filename)
-    log("The file %s does not exist. Will use gabc file", gtex_filename)
-    if gabc_file then
-      local gabc = io.open(gabc_file, 'r')
-      if gabc == nil then
-        err("\n Unable to open %s", gabc_file)
-        return
-      else
-        gabc:close()
-      end
-      local sep = ""
-      local onwindows = os.type == "windows" or
-        string.find(os.getenv("PATH"),";",1,true)
-      if onwindows then
-        sep = "\\"
-      else
-        sep = "/"
-      end
-      local output_dir = base_output_dir..sep..file_dir
-      info(output_dir)
-      if not lfs.exists(output_dir) then
-        if not lfs.exists(base_output_dir) then
-          lfs.mkdir(base_output_dir)
-        end
-        local err,message = lfs.rmkdir(output_dir)
-        if not err then
-          info(message)
-        end
-      end
-      gtex_filename = string.format("%s%s-%s.gtex", output_dir, cleaned_filename,
-          internalversion:gsub("%.", "_"))
-      glog_file = string.format("%s%s-%s.glog", output_dir, cleaned_filename,
-          internalversion:gsub("%.", "_"))
-      compile_gabc(gabc_file, gtex_filename, glog_file, allow_deprecated)
-      gtex_filename = lfs.normalize(gtex_filename)
-      tex.print(string.format([[\input %s\relax]], gtex_filename))
-      return
-    else
-      err("The file %s does not exist", gabc_filename)
-      return
-    end
+    gabc_dir, base = string.match(gabc_file, "(.-)([^\\/]*)$")
   end
-  if not gabc_file then
```

Comment:
This seems to have used the gtex file even if the original gabc file was gone. I didn't keep this case, but it could be easily added back in.

Reply:
We want to keep that case.  A "missing" gabc file may actually be one that we've purposely hidden from the document because we've made manual modifications to the gtex file.

Reply:
Will do